lib/msal-core/samples/react-sample-app/src/AuthProvider.js

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from "react";
+import {
+    msalApp,
+    requiresInteraction,
+    fetchMsGraph,
+    isIE,
+    GRAPH_ENDPOINTS,
+    GRAPH_SCOPES,
+    GRAPH_REQUESTS
+} from "./auth-utils";
+
+// If you support IE, our recommendation is that you sign-in using Redirect APIs
+const useRedirectFlow = isIE();
+
+export default C =>
+    class AuthProvider extends Component {
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                account: null,
+                error: null,
+                emailMessages: null,
+                graphProfile: null
+            };
+        }
+
+        async acquireToken(request, redirect) {
+            return msalApp.acquireTokenSilent(request).catch(error => {
+                // Call acquireTokenPopup (popup window) in case of acquireTokenSilent failure
+                // due to consent or interaction required ONLY
+                if (requiresInteraction(error.errorCode)) {
+                    return redirect
+                        ? msalApp.acquireTokenRedirect(request)
+                        : msalApp.acquireTokenPopup(request);
+                }
+            });
+        }
+
+        async onSignIn(redirect) {
```

Comment:
how does redirect get passed in, the app looks like it just passes the handler to the onClick without binding an option

Reply:
In the render function: 
```jsx
onSignIn={() => this.onSignIn(useRedirectFlow)}
```